Make configure compatible with PHP 7.4

ausi opened the pull request:
This is a followup to #29 using separate config files [as requested](https://github.com/travis-ci/php-src-builder/pull/29#issuecomment-472557912) by @BanzaiMan

nikic committed:
Place ICU_INSTALL_DIR on PKG_CONFIG_PATH

nikic committed:
Add our own libjpeg.pc if it doesn't exist yet

ausi committed:
Adjust configure for changes in PHP 7.4

ausi commented:
The libonig issue still needs to be resolved I think.  

https://github.com/travis-ci/php-src-builder/pull/29#issuecomment-462346556

> the build is now likely to fail due to a missing libonig dependency.
> 
> @BanzaiMan To get an additional dependency, would it be sufficient to install `libonig-dev` in `.travis.yml`, or would this result in a missing shared object dependency when the built binary is used? That is, does something additional need to be done to make sure that the necessary SO is also present in the container/VM in some manner?

joshk reviewed (commented):
Due to the use of : 

```
$TRAVIS_BUILD_DIR/default_configure_options.${RELEASE}
```
in the install files, and the use of `default_configure_options.trusty-master` style configure files, it might be best to use the `custom_configure_options` file and then add them to the default file used.

joshk committed:
improvement to how options are added to default_configure_options

joshk committed:
improvements to cmake and libzip installation

joshk committed:
pear won't install on master/8.0 at the moment

there is an issue with the version of Console_Getopts packaged with PEAR, it packages an old version which uses the each method, which is removed in 8.0

joshk committed:
don't hardcode /home/travis

joshk committed:
make sure the pecl and pear channels are updated

joshk committed:
update the ssl cert

joshk committed:
update the cert before installing pear

joshk committed:
add default configure option for bionic

nikic commented:
Why are you disabling PEAR on master? It should be compatible with PHP 8. At least the Console_GetOpt issue you're referencing was fixed a few weeks ago.

joshk commented:
hey @nikic, I did lots of testing but it seems as though the version of Console_Getopt packaged in PEAR is old and does not include the `each()` fix.

nikic commented on default_configure_options.bionic:
```diff
@@ -0,0 +1,49 @@
+--enable-intl
```
Doesn't this one also need all the version specific variations, at least for baseline, 7.4, master? (Which is also the reason why I still think that #29 is a more scalable solution to this problem.)

joshk replied:
sorry, i borrowed this from the xenial base line default for my testing.

nikic commented:
@joshk Just tested this locally and you are absolutely right. This must be a recent regression as this was working a few days ago :/ I'll try to get this fixed.

joshk commented:
awesome, thanks @nikic 
I spent a lot of time hunting it down and then seeing if I could build an updated release locally, but since PEAR needs PEAR to build PEAR, I was a bit stuck.

joshk commented:
I have to admit that I hijacked this PR a little to test some build system improvements. Once this gets merged in I can create a PR with my tweaks.
(this will make it easier to build other OS versions)

joshk committed:
the * means both the tgz and directory are captured

BanzaiMan commented on bin/compile-extension-memcache:
```diff
@@ -7,7 +7,7 @@ travis_time_start
 if [[ ! $VERSION =~ ^7 && ! $VERSION =~ ^master$ ]]; then
 	pecl download memcache-beta
-	tar zxvf memcache*.tgz && pushd memcache*
+	tar zxvf memcache*.tgz && pushd memcache*/
```
What are these trailing slashes for?

nikic commented:
Forgot to report back here, the go-pear.phar issue has been resolved (again...) so it should be fine to also install on master.

nikic commented:
Also in the meantime `--enable-maintainer-zts` has become `--enable-zts` on master.
